sdk/src/client/api/block_builder/input_selection/requirement/amount.rs

```diff
@@ -153,6 +155,22 @@ impl AmountSelection {
             self.inputs_sum += input.output.amount();
             self.newly_selected_inputs.insert(*input.output_id(), input.clone());
 
+            if input.output.native_token().is_some() {
```

Comment:
Isn't that something we should maybe check before actually selecting this input?

Reply:
We order the outputs based on that before calling this method https://github.com/iotaledger/iota-sdk/blob/8d39fcffdbbba2234c38f704e34a5cfc6e850ed1/sdk/src/client/api/block_builder/input_selection/requirement/amount.rs#L195-L221